### PR TITLE
Feature/implement e2e client connects to node test

### PR DESCRIPTION
**Pull request by tadovas**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 /vendor
 /.env
 testdataoutput
+e2e_tests.log
```
Comment (donce) [inline .gitignore]: newline! :D

Comment (tadovas) [reply .gitignore]: Not again :) fixed.

File: bin/client_package/config/update-resolv-conf [hunk 2]
```diff
@@ -13,6 +13,9 @@
 # copy update-resolv-conf to myst client current directory
 # build/client/ (by default)
 
+#This check helps to avoid running this script in linux (win :)) based environments
+[[ $OSTYPE = "darwin"* ]] || exit 0
```
[the remark this answers is not in the document]
Comment (tadovas) [reply bin/client_package/config/update-resolv-conf]: It's a special case when expanding os type and making string match (wildchar at the end) so yes this expression is valid.

File: bin/client_docker/docker-entrypoint.sh [hunk 3]
```diff
@@ -5,10 +5,13 @@ OS_DIR_CONFIG="/etc/mysterium-client"
 OS_DIR_DATA="/var/lib/mysterium-client"
 OS_DIR_RUN="/var/run/mysterium-client"
 
+mkdir -p $OS_DIR_RUN
```
[the remark this answers is not in the document]
Comment (tadovas) [reply bin/client_docker/docker-entrypoint.sh]: Done

File: e2e/connection_test.go [hunk 4]
```diff
@@ -2,20 +2,76 @@ package e2e
...
 	status, err := tequilApi.Status()
 	assert.NoError(t, err)
 	assert.Equal(t, "NotConnected", status.Status)
+
+	identities, err := tequilApi.GetIdentities()
+	assert.NoError(t, err)
+
+	var identity client.IdentityDTO
+	if len(identities) < 1 {
+		identity, err = tequilApi.NewIdentity("")
```
Comment (Waldz) [inline e2e/connection_test.go]: It's 2nd green path. Should not we have 2 separate tests for that?

Comment (tadovas) [reply e2e/connection_test.go]: It's not green path. It's ensurement that we have available identity to use for connection - test cannot proceed without that.

File: bin/run_e2e_tests [hunk 5]
```diff
@@ -17,17 +17,33 @@ print_error () {
...
+    if [ ! $? -eq 0 ]; then
+        print_error "Db startup failed"
+        cleanup
+        exit 1
+    fi
+
     ${dockerComposeCmd} build
     if [ ! $? -eq 0 ]; then
         print_error "Building docker images failed"
         exit 1
```
Comment (donce) [inline bin/run_e2e_tests]: Cleanup missing.

File: bin/run_e2e_tests [hunk 6]
```diff
@@ -17,17 +17,33 @@ print_error () {
...
 
 setup () {
+    ${dockerComposeCmd} up -d db # start database first - it takes about 10 sec untils db startsup, and otherwise db migration fails
+    if [ ! $? -eq 0 ]; then
+        print_error "Db startup failed"
+        cleanup
+        exit 1
+    fi
+
     ${dockerComposeCmd} build
```
Comment (donce) [inline bin/run_e2e_tests]: What if images are already built, won't this finish instantly while database is not set-up yet?

File: bin/run_e2e_tests [hunk 7]
```diff
@@ -17,17 +17,33 @@ print_error () {
...
     if [ ! $? -eq 0 ]; then
         print_error "Building docker images failed"
         exit 1
     fi
 
+    ${dockerComposeCmd} run --entrypoint bin/db-upgrade discovery
+    if [ ! $? -eq 0 ]; then
+        print_error "Db migration failed"
+        cleanup
+        exit 1
```
Comment (donce) [inline bin/run_e2e_tests]: These 3 lines repeats on each step - we can make a function `fatal(error_message)` to DRY this.

File: cmd/commands/cli/command_cli.go [hunk 8]
```diff
@@ -226,10 +226,8 @@ func (c *Command) proposals() {
 	for _, proposal := range proposals {
 		country := proposal.ServiceDefinition.LocationOriginate.Country
 		var countryString string
-		if country != nil {
-			countryString = *country
-		} else {
-			countryString = "Unknown"
+		if len(country) == 0 {
```
Comment (donce) [inline cmd/commands/cli/command_cli.go]: I'm confusied - Isn't `countryString` left empty if `len(country)` != 0?

Comment (tadovas) [reply cmd/commands/cli/command_cli.go]: Fixed

File: bin/run_e2e_tests (hunk 7 above)
Comment (tadovas) [reply bin/run_e2e_tests]: It's bash - next time

File: tequilapi/client/dto.go [hunk 9]
```diff
@@ -25,14 +27,18 @@ type ProposalDTO struct {
 	ServiceDefinition ServiceDefinitionDTO `json:"serviceDefinition"`
 }
 
+func (p ProposalDTO) String() string {
+	return fmt.Sprintf("Id: %d , Provider: %s, Country: %s", p.ID, p.ProviderID, p.ServiceDefinition.LocationOriginate.Country)
```
Comment (donce) [inline tequilapi/client/dto.go]:
```go
p.ServiceDefinition.LocationOriginate.Country
```
This is very long call chain and it seems to be repeated everywhere we need country. Perhaps service proposal or service definition could have `country` shortcut method?

**Pull request merged**